gemrb/GUIScripts/bg1/CharGenGui.py

```diff
@@ -313,21 +313,27 @@ def setAccept():
 	#set my character up
 	MyChar = GemRB.GetVar ("Slot")
 
+	#reputation
+	# If Rep > 0 then MyChar is an imported character with a saved reputation,
+	# in which case the reputation shouldn't be overwritten.
+	Rep = GemRB.GetPlayerStat (MyChar, IE_REPUTATION)
+
+	if Rep <= 0:
+		AlignID = GemRB.GetPlayerStat (MyChar, IE_ALIGNMENT)
+		TmpTable = GemRB.LoadTable ("repstart")
+		Rep = TmpTable.GetValue (AlignID, 0) * 10
+		GemRB.SetPlayerStat (MyChar, IE_REPUTATION, Rep)
+
+	if MyChar == 1: # only do it once
+		GemRB.GameSetReputation (Rep)
+
```

Comment:
Comparing the bg1 and bg2 code and the tables involved, I think you're getting the wrong values in bg1. Please check the reputation matches the chosen alignment.

CommonTables.Aligns == gemrb/unhardcoded/bg1/aligns.2da
https://gibberlings3.github.io/iesdp/files/2da/2da_tots/repstart.htm

Reply:
Indeed... the reputation is always 10.

Using the same code as for BG2 I get the correct reputations in BG1. So was this always broken?

Reply:
Maybe not always, but possible.

Reply:
And SonarCloud is -- rightfully -- complaining about duplicated code. I'll put the reputation lookup for BG1 and BG2 into gemrb/GUIScripts/GUICommon.py.

Reply:
gemrb/GUIScripts/BGCommon.py is a better place, since it's not useful for the other games.